controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -0,0 +1,198 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package dogstatsd
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"sigs.k8s.io/yaml"
+
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+)
+
+func init() {
+	err := feature.Register(feature.DogStatsDIDType, buildDogStatsDFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildDogStatsDFeature(options *feature.Options) feature.Feature {
+	dogStatsDFeat := &dogStatsDFeature{}
+
+	return dogStatsDFeat
+}
+
+type dogStatsDFeature struct {
+	hostPortEnabled  bool
+	hostPortHostPort int32
+
+	udsEnabled      bool
+	udsHostFilepath string
+
+	originDetectionEnabled bool
+	mapperProfiles         *apicommonv1.CustomConfig
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *dogStatsDFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	dogstatsd := dda.Spec.Features.Dogstatsd
+	if apiutils.BoolValue(dogstatsd.HostPortConfig.Enabled) {
+		f.hostPortEnabled = true
+	}
+	if dogstatsd.HostPortConfig.Port != nil {
+		f.hostPortHostPort = *dogstatsd.HostPortConfig.Port
+	}
+	if apiutils.BoolValue(dogstatsd.UnixDomainSocketConfig.Enabled) {
+		f.udsEnabled = true
+	}
+	f.udsHostFilepath = *dogstatsd.UnixDomainSocketConfig.Path
+	if apiutils.BoolValue(dogstatsd.OriginDetectionEnabled) {
+		f.originDetectionEnabled = true
+	}
+	if dogstatsd.MapperProfiles != nil {
+		f.mapperProfiles = v2alpha1.ConvertCustomConfig(dogstatsd.MapperProfiles)
+	}
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
+			},
```

Comment:
should it always return `IsRequired: true` ?

Reply:
It is for `ConfigureV1` but I missed setting that for `Configure`. I'll change that.